### PR TITLE
[TechDraw] Reimplement Leader Line parent linking

### PR DESCRIPTION
**Pull request by pavltom**
This pull request continues with unification of parent/owner relationship based on PRs #12342  and #12421, this time for Leader Lines. As Leader Line can be also a parent of e.g. Rich Annotations or Symbols, this pull requests allows to transfer the Leader Line to any other Part View present on the page, while keeping all its children, without the need to re-create them.

Because the infrastructure is already present, the major part of this PR is removal of the code duplicating the functionality. This will make future maintenance much easier. Later, I would like to finish the task with Weld Symbols which will have their own PR.

I have also added a standalone commit to general Gui module implementing the Link Property value changes. Currently the dialog tries to display the selected link target in 3D view, but TechDraw's Part Views are not a part of 3D geometry and thus are not present in the 3D view. This change reflects that and brings the correct Page to front when desired link target is clicked. This commit can be merged together with other changes or cherrypicked alone later.

Again, I have tried to test the changes properly, but still there can be some omissions (like Projection Groups last time), so please let me know, if you encounter any issues. Of course any suggestions are also welcome.

### CODE REVIEW

File: src/Gui/DlgPropertyLink.cpp [hunk 1]
```diff
@@ -525,7 +525,13 @@ void DlgPropertyLink::onItemSelectionChanged()
             auto vp = Base::freecad_dynamic_cast<Gui::ViewProviderDocumentObject>(
                     doc->getViewProvider(obj));
             if(vp) {
-                doc->setActiveView(vp, Gui::View3DInventor::getClassTypeId());
+                MDIView *view = vp->getMDIView();
```
Comment (WandererFan) [inline src/Gui/DlgPropertyLink.cpp:528]: // handle links that point to objects not found in 3d view?

**Comment by WandererFan** [comment]
The leader line stuff looks good.

The code for  the Link change makes sense in that not all MDIViews are View3dInventor, but I'm not sure I've actually exercised the new code.   Can you give a little script to follow?

**Comment by pavltom** [comment]
Here are few steps to hit the new DlgPropertyLink code:

1) Create some geometry (Sketch, Primitive, etc.)
2) In TechDraw, add a new Page, add two Part Views of the goemetry above, lets name them PartView1, PartView2
3) Add arbitrary Leader Line to PartView1
4) Make sure Tree View is not synchronized with 3D view automatically (this means the General - Selection - Auto switch to the 3D view... preference is not checked)
5) Select the Leader Line and click on the dots in the Leader Parent field of the Data tab to change the parent and invoke the "Link" dialog
6) Now select PartView2 in the list. The current implementation switches to View3dInventor (which does not display any of the Part Views) while the proposed change brings up the page with PartView2 and highlights it.

I guess the essential part is the Auto switch... prefference to be turned off. If it is on, everything works as expected right away.

**Comment by WandererFan** [comment]
Got it.  Thank you.

**Pull request merged**